anchore_engine/db/db_catalog_image_docker.py

```diff
@@ -158,29 +147,34 @@ def get_tag_histories(session, userId, registries=None, repositories=None, tags=
 
     qry = session.query(*select_fields).join(CatalogImage, and_(CatalogImageDocker.userId == CatalogImage.userId, CatalogImageDocker.imageDigest == CatalogImage.imageDigest)).filter(CatalogImage.userId==userId).order_by(*order_by_fields)
 
-    for field, filters in [(CatalogImageDocker.registry, registries), (CatalogImageDocker.repo, repositories), (CatalogImageDocker.tag, tags)]:
-        if filters:
-            wildcarded = []
-            exact = []
-            for r in filters:
-                if r.strip() == '*':
-                    continue
-
-                if '*' in r:
-                    wildcarded.append(r)
-                else:
-                    exact.append(r)
-
-            conditions = []
-            if wildcarded:
-                for w in wildcarded:
-                    conditions.append(field.like(w.replace('*', '%')))
-
-            if exact:
-                conditions.append(field.in_(exact))
-
-            if conditions:
-                qry = qry.filter(or_(*conditions))
+    query_tuples = [(CatalogImageDocker.registry, registries),
+                    (CatalogImageDocker.repo, repositories),
+                    (CatalogImageDocker.tag, tags)]
+    for field, filters, exclude in query_tuples:
+        if not filters:
```

Comment:
I'm not quite sure this code is doing the right thing. The iterator in the for loop should iterate over each of the tuples, but it looks like its expecting 3 values instead of 2. Is the 'exclude' necessary here? It doesn't look like it's used at all in the loop.

Reply:
You were right, it was left over when I was mulling over my strategy for applying the exclusion. I've removed it now.